File: cases/taint/go/direct-negative/direct_flow.go
package dataflowbench

func dfb_source() string { // DFB-SOURCE: direct-input
	return "tainted"
}

func dfb_sink(value string) {} // DFB-SINK: direct-sink

func run() {
	dfb_source()
	dfb_sink("clean")
}

File: cases/taint/scala/direct-negative/DirectFlow.scala
package dataflowbench

object DirectFlow {
  def dfb_source(): String = { // DFB-SOURCE: direct-input
    "tainted"
  }

  def dfb_sink(value: String): Unit = {} // DFB-SINK: direct-sink

  def run(): Unit = {
    dfb_source()
    dfb_sink("clean")
  }
}

